Account/FxAPushMessageHandler.swift

```diff
@@ -60,76 +60,88 @@ extension FxAPushMessageHandler {
         }
 
         // return handle(plaintext: string)
-        let deferred = PushMessageResult()
-        RustFirefoxAccounts.reconfig(prefs: profile.prefs).uponQueue(.main) { accountManager in
-            accountManager.deviceConstellation()?.processRawIncomingAccountEvent(pushPayload: string) {
-                result in
-                guard case .success(let events) = result, let firstEvent = events.first else {
-                    let err: PushMessageError
-                    if case .failure(let error) = result {
-                        err = PushMessageError.messageIncomplete(error.localizedDescription)
-                    } else {
-                        err = PushMessageError.messageIncomplete("empty message")
-                    }
-                    deferred.fill(Maybe(failure: err))
-                    return
-                }
-                if events.count > 1 {
-                    // Log to the console for debugging release builds
-                    os_log(
-                        "%{public}@",
-                        log: OSLog(subsystem: "org.mozilla.firefox",
-                                   category: "firefoxnotificationservice"),
-                        type: OSLogType.debug,
-                        "Multiple events arrived, only handling the first event.")
-                }
-                switch firstEvent {
-                case .commandReceived(let deviceCommand):
-                    switch deviceCommand {
-                    case .tabReceived(_, let tabData):
-                        let title = tabData.entries.last?.title ?? ""
-                        let url = tabData.entries.last?.url ?? ""
-                        let message = PushMessage.commandReceived(tab: ["title": title, "url": url])
-                        if let json = try? accountManager.gatherTelemetry() {
-                            let events = FxATelemetry.parseTelemetry(fromJSONString: json)
-                            events.forEach { $0.record(intoPrefs: self.profile.prefs) }
+        let deferred = PushMessageResults()
+        // Reconfig has to happen on the main thread, since it calls `startup`
+        // and `startup` asserts that we are on the main thread. Otherwise the notification
+        // service will crash.
+        DispatchQueue.main.async {
```

Comment:
I'm mildly surprised send tab even works in prod correctly. I had to add this because my device couldn't receive send tabs properly, and after debugging I found that the culprit is that the NotificationService on my device is running a background thread when receiving messages

This causes it to crash and deliver a default notification instead of the actual tab

I'm assuming in production builds (and in beta I tested to make sure) the notification service runs on the main thread somehow? regardless dispatching this onto the main thread fixed local push notification for me on my iPhone